WcaOnRails/app/models/competition.rb

```diff
@@ -473,6 +473,12 @@ def events_with_rounds_with_results
     end
   end
 
+  private def light_results_from_relation(relation)
+    ActiveRecord::Base.connection.execute(relation.to_sql).each(as: :hash).map do |r|
+      LightResult.new(r)
```

Comment:
Interesting. Can you explain the reason for creating a LightResult rather than a regular Result?